### PR TITLE
ui: Add content placeholders

### PR DESCRIPTION
**Pull request by Djo1e**
This PR replaces loaders with content placeholders. By doing this we remove jitter and give the user a visual hint. This is especially useful in the `Send` form, where the user can see what information needs to be entered.

I've opted to use the `Placeholder` component from `react-bootstrap` instead of introducing an additional lib.

**Before:**
https://www.loom.com/share/fc39755d2a1f4f288cb7419470cf08ba

**After:**
https://www.loom.com/share/ac93b1ac92994390a51ef49a5f490b41

**How to test:**
Run locally, make sure all buttons are disabled and that no action can be taken before loading is complete. 

I'm fairly new to the codebase/app so please test carefully. All suggestions are welcome!

### CODE REVIEW

File: src/components/Send.jsx
Comment (theborakompanioni) [reply src/components/Send.jsx:495]: Feels that this could be removed by someone not knowing the impacts in the future.  Do you think the whole content should be wrapped in `<></>` then?

**Comment by theborakompanioni** [comment]
This is a really nice visual improvement. The current loading behaviour always felt clunky. The screens are looking great already. Cannot wait till this is ready for review!

**Comment by dnlggr** [comment]
Awesome thanks for this PR @Djo1e! 🙏 🙌 

Can I mark it as ready for review?

File: src/components/Send.jsx
Comment (Djo1e) [reply src/components/Send.jsx]: Good point. Changed!

File: src/components/CurrentWalletAdvanced.tsx
Comment (dnlggr) [inline src/components/CurrentWalletAdvanced.tsx:72]: Are you planning on reusing `LoadingAccounts` anywhere else? If not we could think about inlining it here similar to the placeholders in the other components.

File: src/components/LoadingAccounts.css
Comment (dnlggr) [inline src/components/LoadingAccounts.css:4]: Let's use [CSS Modules](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/) when putting component CSS into separate files.

File: src/components/LoadingAccounts.tsx
Comment (dnlggr) [inline src/components/LoadingAccounts.tsx:11]:
```suggestion
        <rb.Placeholder key={index} as="div" animation="wave" style={{ marginBottom: 6 }}>
```

Let's be consistent in the animation we use. I don't have a strong preference over "wave" or "glow" but whatever we pick, let's make it consistent across the app.

Comment (dnlggr) [inline src/components/LoadingAccounts.tsx:11]:
```suggestion
        <rb.Placeholder key={index} as="div" animation="glow">
```

Let's use the CSS from the CSS module here.

File: src/components/Send.css
Comment (dnlggr) [inline src/components/Send.css:132]:
```suggestion
```

This is not used anywhere I believe.

File: src/components/Send.css
Comment (dnlggr) [inline src/components/Send.css:128]:
```suggestion
.send .input-loader {
  height: 3.5rem;
}
```

Since this file is not a CSS Module, we have to be careful about naming clashes.

**Review by dnlggr — COMMENTED** [review]
Thanks for this PR @Djo1e! Great job, it's looking really good already. Great visual improvement compared to before. 🎉 👏 

I left some comments inline. Let me know what you think or if you need further clarification on something.

On the receive screen,  let's also disable the "Settings" button while the content is loading:

Screenshot 2022-04-06 at 11 16 28

---

Final thing: The `CurrentWalletMagic` screen could use some placeholders as well. This is the first screen users will see first. Would be great if you could you take a look at that as well. 🙏

**Comment by Djo1e** [comment]
@dnlggr Thanks for the review! I have already addressed some of them, but will need to clean it up a little bit. I'll probably finish this over the weekend when I have more time. I added the content placeholder for the `CurrrentWalletMagic` screen, but since we won't know how many wallets are there until loading is finished, I'm not sure how many placeholders to add. I tried something, but it does't look that good if there are less wallets than the loaders. See https://www.loom.com/share/4eb1f4573e4b476491a4820561080dd2

**Comment by dnlggr** [comment]
Thanks for the update @Djo1e! No worries take your time.

For the `CurrentWalletMagic` I actually meant this screen:

Screenshot 2022-04-07 at 12 31 22

You can toggle it to be the default wallet screen in settings:

Screenshot 2022-04-07 at 12 31 30

**Comment by editwentyone** [comment]
I like it so far :) is there a chance to add subtle animated dots like "…" per field? 
maybe its a bit much, but I would show the user more clear that something is "on the way"

**Comment by dnlggr** [comment]
> I like it so far :) is there a chance to add subtle animated dots like "…" per field?
maybe its a bit much, but I would show the user more clear that something is "on the way"

The placeholders have a built in loading animation. It's just not very well visible with the current color choice. Maybe we can find a better color for the placeholders @Djo1e?

Example with a (too) dark color just to show the animation:

https://user-images.githubusercontent.com/10026790/162457038-f6b94540-089d-4e02-aae4-0ecd5338380f.mov

**Comment by editwentyone** [comment]
keep it as it is, I would love to see it in action on my screen and then we could tweak it further if needed. also lets see what other feedback will come back

File: src/components/CurrentWalletMagic.jsx
Comment (Djo1e) [inline src/components/CurrentWalletMagic.jsx:81]: Typescript was complaining

File: src/components/ExtendedLink.tsx
Comment (Djo1e) [inline src/components/ExtendedLink.tsx:7]: I've extended the `Link` component since it doesn't have a disabled state internally

File: src/components/CurrentWalletAdvanced.tsx
Comment (Djo1e) [reply src/components/CurrentWalletAdvanced.tsx]: Changed

File: src/components/LoadingAccounts.tsx
Comment (Djo1e) [reply src/components/LoadingAccounts.tsx]: Good catch, changed.

File: src/components/Send.css
Comment (Djo1e) [reply src/components/Send.css]: Removed

File: src/components/Send.css
Comment (Djo1e) [reply src/components/Send.css]: Done. I've moved all css that isn't global into an appropriate module.

**Review by dnlggr — DISMISSED** [review]
Thanks for the update @Djo1e! 🎉 

Visually looks great now! Please fix the `undefined` in the Balance component, though. After that we should be good to merge. 👍

File: src/components/Balance.jsx
Comment (Djo1e) [reply src/components/Balance.jsx:79]: Done!

File: src/components/Balance.jsx
Comment (Djo1e) [reply src/components/Balance.jsx]: Fixed

File: src/components/CurrentWalletMagic.jsx
Comment (Djo1e) [reply src/components/CurrentWalletMagic.jsx]: It should - added

File: src/components/ExtendedLink.tsx
Comment (Djo1e) [reply src/components/ExtendedLink.tsx]: removed

**Comment by theborakompanioni** [comment]
Two things that were noticed:
- Sprites "node", "receive", "send" were not adopted
- flickering when loading a new address on "Receive" screen

I guess these are rather minor changes, so we could also fix them later.
Waiting for an answer as you might want to fix them yourself.
@dnlggr What do you think, are we ready to merge?

Thank you @Djo1e :pray:

**Comment by Djo1e** [comment]
@theborakompanioni I've added the missing sprites and fixed the flickering. Thanks for taking the time to review (@dnlggr as well!)

**Review by theborakompanioni — APPROVED** [review]
I guess some really tiny stuff still needs to be adjusted, e.g. "send" and "receive" sprites in the buttons in `CurrentWalletMagic` screen are missing - but we can address that later.

lgtm!

**Comment by dnlggr** [comment]
Thanks @theborakompanioni and @Djo1e. Yeah I agree, we can address little things separately. 

Thanks for the contribution @Djo1e and thanks for taking over the review @theborakompanioni! 🎉

**Pull request merged**
